utils/lint_test_cases.py

```diff
@@ -0,0 +1,924 @@
+
+    # Filename patterns that indicate specific backends
+    FILENAME_PATTERNS = {
+        Backend.AZURE: [
+            r"azure",
+        ],
+        Backend.GCE: [
+            r"gce",
+            r"custom-d3",
+        ],
+        Backend.DOCKER: [
+            r"docker",
+        ],
+        Backend.BAREMETAL: [
+            r"baremetal",
+        ],
+        Backend.K8S_LOCAL_KIND: [
+            r"scylla-operator",
+            r"kubernetes",
+        ],
+        Backend.K8S_EKS: [
+            r"scylla-operator",
+            r"eks",
+        ],
+        Backend.K8S_GKE: [
+            r"scylla-operator",
+            r"gke",
+        ],
+    }
+
+    # Content keys that indicate specific backends
+    CONTENT_BACKEND_KEYS = {
+        Backend.AWS: ["region_name", "instance_type_db", "ami_id_db_scylla"],
+        Backend.AZURE: ["azure_region_name", "azure_instance_type_db", "azure_image_db"],
+        Backend.GCE: ["gce_datacenter", "gce_instance_type_db", "gce_image_db"],
+        Backend.DOCKER: ["docker_image", "docker_network"],
+        Backend.BAREMETAL: ["db_nodes_public_ip", "db_nodes_private_ip", "s3_baremetal_config"],
+        Backend.K8S_LOCAL_KIND: ["k8s_n_scylla_pods_per_cluster", "k8s_scylla_operator_docker_image", "k8s_loader_run_type"],
+        Backend.K8S_EKS: ["k8s_n_scylla_pods_per_cluster", "k8s_scylla_operator_docker_image", "k8s_loader_run_type", "k8s_minio_storage_size"],
+        Backend.K8S_GKE: ["k8s_n_scylla_pods_per_cluster", "k8s_scylla_operator_docker_image", "k8s_loader_run_type", "k8s_minio_storage_size"],
+    }
+
+    def __init__(self, test_cases_dir: Optional[Path] = None):
+        if test_cases_dir is None:
+            # Try to find test-cases directory relative to this file
+            script_dir = Path(__file__).resolve().parent
+            project_root = script_dir.parent
+            self.test_cases_dir = project_root / "test-cases"
+        else:
+            self.test_cases_dir = test_cases_dir
+
+        # Store project root for relative path display
+        self.project_root = Path(__file__).resolve().parent.parent
+
+    def _run_yaml_test(self, backend: str, full_path: str, env: Dict[str, str]) -> Tuple[bool, List[str]]:
+        """
+        Run YAML test validation directly using SCTConfiguration.
+
+        This is a direct port of the _run_yaml_test function from sct.py,
+        avoiding subprocess calls.
+
+        Args:
+            backend: Backend name (aws, gce, azure, etc.)
+            full_path: Full path to the YAML test case file
+            env: Environment variables to set
+
+        Returns:
+            Tuple of (error: bool, output: List[str])
+        """
+        output = []
+        error = False
+        output.append(f'---- linting: {full_path} -----')
+
+        try:
+            # Clear and set environment variables
+            for key in os.environ.keys():
+                if key.startswith('SCT_'):
+                    del os.environ[key]
```

Comment:
this happen in a cleanup, ideally in some context manager. We should also have a better option how to parse these into SCTConfiguration without os.environ